lib/structures/VoiceState.ts

```diff
@@ -49,25 +48,25 @@ export default class VoiceState extends Base {
         this.selfStream = false;
         this.selfVideo = false;
         this.suppress = false;
-        this.user = client.users.get(this.id)!;
         this.update(data);
     }
 
     protected update(data: Partial<RawVoiceState>) {
-        if (data.channel_id !== undefined) {
-            if (data.channel_id === null) this.channel = null;
-            else {
-                const ch = this.client.getChannel<VoiceChannel | StageChannel>(data.channel_id);
-                if (!ch) this.client.emit("warn", `Missing channel for VoiceState ${this.id}`);
-                else this.channel = ch;
-            }
-        }
         if (data.deaf !== undefined) this.deaf = data.deaf;
         if (data.guild_id !== undefined) {
+            this.guildID = data.guild_id;
             const guild = this.client.guilds.get(data.guild_id);
             if (!guild) this.client.emit("warn", `Missing guild for VoiceState ${this.id}`);
             else this.guild = guild;
         }
+        if (data.channel_id !== undefined) {
+            if (data.channel_id === null) this.channel = null;
+            else {
+                const ch = this.guild?.channels.get<VoiceChannel | StageChannel>(data.channel_id);
```

Comment:
channels.get does not accept a Generic, what's the issue with client.getChannel?